### PR TITLE
Adding CREATE support

### PR DESCRIPTION
**Pull request by Guttz**
First step into the implementation of https://github.com/HTTP-APIs/hydra-python-agent/issues/101, a full featured client

**Also aligning develop branch which was behind**

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.

### Description

Now can push to hydrus using the command: `create in <collection_endpoint> <json-ld object>`

### Change logs
Added insert in CollectionEndpoints class
Added create interpreter for cli
Added insert func in querying mech

#### Testing 
For testing this with the drone API just run:
`create in DroneCollection {"@type":"Drone","DroneState" : "Simplified state", "name" : "Smart Drone", "model" : "Hydra Drone", "MaxSpeed" : "999", "Sensor" : "Wind"}
`

### CODE REVIEW

File: hydra_agent/collections_endpoint.py [hunk 1]
```diff
@@ -23,17 +24,43 @@ def fetch_data(self, new_url):
...
             return RequestError("error")
         except ValueError as e:
-            logger.info("value error:",e)
+            logger.info("Request Value Error:", e)
             return RequestError("error")
         else:
             return json.loads(response.read().decode('utf-8'))
 
+    def put_data(self, collection_url, new_object):
+        """Fetching data from the server
```
Comment (xadahiya) [inline hydra_agent/collections_endpoint.py]: Update the Docstrings

File: hydra_agent/collections_endpoint.py [hunk 2]
```diff
@@ -201,8 +224,38 @@ def load_from_server(
                 redis_connection.delete(key)
         # save the new data.
         self.redis_graph.commit()
-#        for node in self.redis_graph.nodes.values():
-#            print("\n",node.alias)
+        # for node in self.redis_graph.nodes.values():
+        # print("\n",node.alias)
+
+    def insert_server(
```
Comment (xadahiya) [inline hydra_agent/collections_endpoint.py]: insert_server is not very descriptive please use better function names. Also, update the doctrings

File: hydra_agent/querying_mechanism.py [hunk 3]
```diff
@@ -526,110 +539,132 @@ def user_query(self, query):
...
+        elif "create" in query:
+            query = query.replace("create", "").strip()
+            if "in" in query:
+                query = query.replace("in", "").strip()
+                collection = query.split(" ", 1)[0]
+                new_object = query.replace(collection, "").strip()
+                self.members = CollectionmembersQuery(self.api_doc,
+                                                        self.url,
+                                                        self.graph)
+                response = self.members.insert_server('DroneCollection', '{"@type":"Drone", "DroneState" : "dronestate", "name" : "Drone3", "model" : "model", "MaxSpeed" : "speed", "Sensor" : "sensor"}')
```
Comment (xadahiya) [inline hydra_agent/querying_mechanism.py]: This is hardcoded. Perhaps you forgot to update this

File: hydra_agent/querying_mechanism.py [hunk 4]
```diff
@@ -192,6 +192,19 @@ def get_members(self, query):
             print(self.connection.smembers("fs:endpoints"))
             return self.data_from_server(endpoint)
 
+    def insert_server(self, endpoint, new_object):
```
Comment (xadahiya) [inline hydra_agent/querying_mechanism.py]: What's the point of this function? You can directly call `insert_server` where you're calling this function.

File: hydra_agent/querying_mechanism.py [hunk 5]
```diff
@@ -526,110 +539,132 @@ def user_query(self, query):
...
+                logger.info("Incorrect query: Use 'help' to know about\
+                            querying format")
                 return None
+        elif "create" in query:
+            query = query.replace("create", "").strip()
+            if "in" in query:
+                query = query.replace("in", "").strip()
+                collection = query.split(" ", 1)[0]
+                new_object = query.replace(collection, "").strip()
+                self.members = CollectionmembersQuery(self.api_doc,
```
Comment (xadahiya) [inline hydra_agent/querying_mechanism.py]: Not sure you need to use `self.members` here, each query is run one time so using a simple variable `members` will do.

**Comment by Mec-iS** [comment]
As discussed in the call, this is not the right approach. We don't need to fix the actual scripts but come out with a design that can embed the current code we have. I close this PR and open a proper issue for the job.
Let's start over from this ---> https://github.com/HTTP-APIs/hydra-python-agent/issues/101#issuecomment-497685202

File: hydra_agent/collections_endpoint.py (hunk 2 above)
Comment (Guttz) [reply hydra_agent/collections_endpoint.py]: I fixed all the other comments on my local repo. Thanks for the notes @xadahiya 

Regarding the function names, I tried to follow the previous standard. But guess I'll refactor them with the new design we discussed on the call

**Comment by motormouthdoda** [comment]
I think you should start with improving the current GET mechanism rather than designing a new mechanism for PUSH/PUT. I think it'll help you avoid any contingencies that can occur later while designing other requests. Moreover improving the GET request will build up a base for other features as well (like advanced querying, better Redis integration, better structure for storing all the Hydra resources)

**Comment by Guttz** [comment]
Yes, we're going with a design document and what will probably lead to refactoring the existing GET mechanism into this new architecture. Lorenzo shared more details at https://github.com/HTTP-APIs/hydra-python-agent/issues/101#issuecomment-497685202.

Next week I should be sharing a doc/design where you're welcome to give your inputs @motormouthdoda

**Pull request closed without merging**